pennylane/measurements/classical_shadow.py

```diff
@@ -314,6 +315,115 @@ def process(self, tape, device):
 
         return qml.math.cast(qml.math.stack([outcomes, recipes]), dtype=np.int8)
 
+    def process_state_with_shots(
+        self, state: Sequence[complex], wire_order: Wires, shots: int, rng=None
+    ):
+        """Process the given quantum state with the given number of shots
+
+        Args:
+            state (Sequence[complex]): quantum state
+            wire_order (Wires): wires determining the subspace that ``state`` acts on; a matrix of
+                dimension :math:`2^n` acts on a subspace of :math:`n` wires
+            shots (int): The number of shots
+            rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+                seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+                If no value is provided, a default RNG will be used.
+
+        Returns:
+            tensor_like[int]: A tensor with shape ``(2, T, n)``, where the first row represents
+            the measured bits and the second represents the recipes used.
+        """
+        wire_map = {w: i for i, w in enumerate(wire_order)}
+        mapped_wires = [wire_map[w] for w in self.wires]
+        n_qubits = len(mapped_wires)
+        num_dev_qubits = len(state.shape)
+
+        # seed the random measurement generation so that recipes
+        # are the same for different executions with the same seed
+        recipe_rng = np.random.RandomState(self.seed)
+        recipes = recipe_rng.randint(0, 3, size=(shots, n_qubits))
```

Comment:
Is there any interest in making these recipes themselves trainable/differentiable?

Reply:
Hmmm, but couldn't one also try to decompose the measurement on each qubit into a linear combination of single-qubit paulis? Similar to https://github.com/matteoacrossi/adapt_ic-povm.

Reply:
Sure, something like that could be possible. It requires that the recipes aren't randomly generated but instead carried over each iteration (or alternatively allow the user to specify them). More discussion is needed on this interface, so it's way outside the scope of this PR.

Reply:
Makes sense. Sounds like that could be a separate PR to add something like `TrainableShadowMP` or some such.